src/OpenTelemetry.Exporter.Zipkin/ZipkinTraceExporter.cs

```diff
@@ -131,20 +145,51 @@ internal ZipkinSpan GenerateSpan(SpanData otelSpan, ZipkinEndpoint defaultLocalE
                 spanBuilder.ParentId(this.EncodeSpanId(otelSpan.ParentSpanId));
             }
 
+            string peerService = null; // RemoteEndpoint.ServiceName primary.
+            string peerHostName = null; // RemoteEndpoint.ServiceName alternative.
+            string peerAddress = null; // RemoteEndpoint.ServiceName alternative.
+            string httpHost = null; // RemoteEndpoint.ServiceName for Http.
+            string dbInstance = null; // RemoteEndpoint.ServiceName for Redis.
             foreach (var label in otelSpan.Attributes)
             {
-                spanBuilder.PutTag(label.Key, label.Value.ToString());
+                string key = label.Key;
+                string strVal = label.Value.ToString();
+
+                if (strVal != null)
+                {
+                    if (key == "peer.service")
```

Comment:
A switch statement would be more concise.

Reply:
@MikeGoldsmith Went with a case-insensitive dictionary.